judgments/forms/fields.py

```diff
@@ -0,0 +1,98 @@
+from calendar import monthrange
+from datetime import date
+from typing import Literal
+
+from crispy_forms_gds.fields import DateInputField
+from django import forms
+from django.core.validators import RegexValidator
+
+from .validators import validate_year_is_within_sensible_range
+
+
+class DateRangeInputField(DateInputField):
+    """
+    A custom version of the gds DateInputField which fixes the issue
+    preventing some of the sub-fields being set as not required.
+
+    The expected use case for this is to have two of these fields on a given
+    form, one being date from and one being date to.
+
+    If a day or month is not provided, default to the minimum or maximum
+    date value depending on whether it is to/from.
+
+    If year is missing raise a validation error.
+
+    e.g. - value of only year provided as from 2010, default to 01/01/2010
+         - value of only year provided as to 2010, default to 31/12/2010
+         - value of year and month provded as to 02/2010, default to 28/02/2010
+    """
+
+    DateType = Literal["from", "to"]
+
+    date_type: DateType
+
+    def __init__(self, date_type, **kwargs):
+        self.date_type = date_type
+        # Define one message for all fields.
+        error_messages = {
+            "required": "Enter the day, month and year",
+            "incomplete": "Enter the day, month and year",
```

Comment:
Aah the behaviour we have at the moment is that a user can leave the month and or day blank, and it will search for the entirety of that year/month: for instance if you put 2003 to 2005 and leave months and years blank, it will default to "2003-01-01" to "2005-12-31": this is by design. Is that same behaviour replicated here or are we requiring the user to input something in all three fields?

Reply:
Sorry! Didn't see the below comment!